br/pkg/lightning/restore/restore.go

```diff
@@ -1508,15 +1544,33 @@ func (rc *Controller) restoreTables(ctx context.Context) (finalErr error) {
 				return errors.Trace(err)
 			}
 
-			wg.Add(1)
-			select {
-			case taskCh <- task{tr: tr, cp: cp}:
-			case <-ctx.Done():
-				return ctx.Err()
+			allTasks = append(allTasks, task{tr: tr, cp: cp})
+
+			if len(cp.Engines) == 0 {
+				for _, fi := range tableMeta.DataFiles {
+					totalDataSizeToRestore += fi.FileMeta.FileSize
+				}
+			} else {
+				for _, eng := range cp.Engines {
+					for _, chunk := range eng.Chunks {
+						totalDataSizeToRestore += chunk.Chunk.EndOffset - chunk.Chunk.Offset
```

Comment:
seems when task is paused and resumed, we'll use unfinished bytes to calculate progress. This will display a 0% progress when resuming, which makes me think that the resuming is failed and we're starting from scratch 🤔

Reply:
it's consistent to previous impl, cal progress from current checkpoint, not from the datafile begin.